File: examples/java/documentation/src/test/java/com/company/app/MainTest.java
package com.company.app;

// DOCS_BLOCK_START:unit-tests
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertTrue;
import com.hashicorp.cdktf.Testing;
import com.hashicorp.cdktf.TerraformStack;
import imports.docker.container.Container;
import imports.docker.image.Image;

import com.mycompany.app.MyApplicationsAbstraction; // Could be a class extending from Construct
import java.util.HashMap;

public class MainTest {

    private final TerraformStack stack = new TerraformStack(Testing.app(), "stack");
    private final MyApplicationsAbstraction appAbstraction = new MyApplicationsAbstraction(stack, "resource");
    private final String synthesized = Testing.synth(stack);

    @Test
    void shouldContainContainer() {
        assertTrue(Testing.toHaveResource(synthesized, Container.TF_RESOURCE_TYPE) );
    }

    @Test
    void shouldUseUbuntuImage() {
        assertTrue(Testing.toHaveResourceWithProperties(synthesized, Image.TF_RESOURCE_TYPE, new HashMap<String, Object>() {
            {
                put("name", "ubuntu:latest");
            }
        }) );
    }
}
// DOCS_BLOCK_END:unit-tests

File: examples/typescript/documentation/__tests__/main-test.ts
// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
// SPDX-License-Identifier: MPL-2.0
// DOCS_BLOCK_START:unit-tests
import { Testing } from "cdktf";
import { Image } from "../.gen/providers/docker/image";
import { Container } from "../.gen/providers/docker/container";
import { MyApplicationsAbstraction } from "../unit-tests"; // Could be a class extending from Construct

export class something {}

describe("Unit testing using assertions", () => {
  beforeAll(async () => {});
  it("should use an docker provider", () => {
    expect(
      Testing.synthScope((scope) => {
        new MyApplicationsAbstraction(scope, "test");
      })
    ).toHaveResource(Container);
  });
  it("should use an docker provider with properties", () => {
    expect(
      Testing.synthScope((scope) => {
        new MyApplicationsAbstraction(scope, "test");
      })
    ).toHaveResourceWithProperties(Image, { name: "ubuntu:latest" });
  });
});
// DOCS_BLOCK_END:unit-tests
